src/benchmark.ts

```diff
@@ -0,0 +1,115 @@
+import { CommitFlags, CommitTransfer, CreateAccount, createClient, CreateTransfer } from '.'
+
+const REQUEST_BATCH_SIZE = 10000
+const MAX_TRANSFERS = 600000
+
+const client = createClient({
+  cluster_id: 0x0a5ca1ab1ebee11en,
+  replica_addresses: ['3001']
+})
+
+const accountA: CreateAccount = {
+  id: 137n,
+  user_data: 0n,
+  reserved: Buffer.alloc(48, 0),
+  code: 0,
+  flags: 0,
+  unit: 1,
+  debits_accepted: 0n,
+  debits_reserved: 0n,
+  credits_accepted: 0n,
+  credits_reserved: 0n,
+  timestamp: 0n,
+}
+
+const accountB: CreateAccount = {
+  id: 138n,
+  user_data: 0n,
+  reserved: Buffer.alloc(48, 0),
+  code: 0,
+  flags: 0,
+  unit: 1,
+  debits_accepted: 0n,
+  debits_reserved: 0n,
+  credits_accepted: 0n,
+  credits_reserved: 0n,
+  timestamp: 0n,
+}
+
+// pre-allocate 1 million transfers and commits
+console.log(`pre-allocating ${MAX_TRANSFERS} transfers and commits...`)
+const transfers: CreateTransfer[] = []
+const commits: CommitTransfer[] = []
+
+for (let i = 0; i < MAX_TRANSFERS; i++) {
```

Comment:
When last I checked, V8 has Arrays being contiguous for up to 10,000 elements. Beyond that, it converts the JS Array to be backed by a sparse hash map and performance degrades significantly.

Let's rather have an Array of Arrays (each with at most 10,000 events). That should help with the overall setup time of the benchmark, and also the runtime and memory usage.

Reply:
Once we have that in place, it would also be interesting to see what happens to memory usage and runtime as we cross the rubicon from 10,000 to 10,001 events per batch or from 9,999 to 10,000 events per batch (in case it's actually 10,000 inclusive that triggers the remapping).